collated/main/A0139942W.md

```diff
@@ -0,0 +1,830 @@
+
+		LocalDateTime currStartTime = timeList.get(FIRST_EVENT_INDEX).getKey();
+		LocalDateTime currEndTime = timeList.get(FIRST_EVENT_INDEX).getValue();
+		
+		
+		if (timeList.size() == ONE_EVENT_ON_THAT_DAY) {
+		    return freetimeForOneEvent(currStartTime , currEndTime, onThatDay, sb);
+	
+		}
+	
+		return freetimeForMutipleEvents(currStartTime, currEndTime, onThatDay, sb);
+
+
+	}
+	
+	private String freetimeForOneEvent(LocalDateTime startTime, LocalDateTime endTime, LocalDateTime onThatDay, StringBuilder sb) {
+		int startTimeDay = startTime.getDayOfMonth();
+		int endTimeDay = endTime.getDayOfMonth();
+		int sameDay = onThatDay.getDayOfMonth();
+		LocalDateTime tempEndTime = endTime;
+		sb.append(String.format(DEFAULT_STARTING_MESSAGE, onThatDay.format(dateFormat)));
+		
+		if (checkOnGoingEvent(startTimeDay, endTimeDay, sameDay)) {
+			sb.append(String.format(ONGOING_EVENT_MESSAGE, startTime.format(datetimeFormat), endTime.format(datetimeFormat)));
+			return sb.toString();
+		}
+		
+		if (isTimeAfterActiveHour(activeHourStart, startTime) && isTimeBeforeActiveHour(activeHourEnd, endTime)) {
+			return sb.append(NO_FREE_TIME_MESSAGE).toString();
+		}
+		
+		if (isTimeBeforeActiveHour(activeHourStart, startTime)) {
+			sb.append(String.format(BETWEEN_EVENT_MESSAGE, activeHourStart.format(hourFormat), startTime.format(hourFormat)));
+		}
+		
+		if (isTimeAfterActiveHour(activeHourStart, endTime)) {
+			tempEndTime = activeHourStart;
+		}
+
+		if (isTimeAfterActiveHour(activeHourEnd, endTime)) {
+			sb.append(String.format(BETWEEN_EVENT_MESSAGE, tempEndTime.format(hourFormat), activeHourEnd.format(hourFormat)));
+			return sb.toString();
+		}
+
+		return sb.toString();
+		
+
+	}
+	
+
+	
+	private String freetimeForMutipleEvents(LocalDateTime startTime, LocalDateTime endTime,LocalDateTime thatDay, StringBuilder sb) {
+		
+		int same_day = thatDay.getDayOfMonth();
+		sb.append(String.format(DEFAULT_STARTING_MESSAGE, thatDay.format(dateFormat)));
+		LocalDateTime currentStartTime = startTime;
+		LocalDateTime currentEndTime = endTime;
+		
+		int startTimeDay = currentStartTime.getDayOfMonth();
+		int endTimeDay = currentEndTime.getDayOfMonth();
+		
+		if (checkOnGoingEvent(startTimeDay, endTimeDay, same_day)) {
+			sb.append(String.format(ONGOING_EVENT_MESSAGE, currentStartTime.format(datetimeFormat), currentEndTime.format(datetimeFormat)));
+			return sb.toString();
+		}
+		
+		if (isTimeBeforeActiveHour(activeHourStart, currentStartTime)) {
+			sb.append(String.format(BETWEEN_EVENT_MESSAGE, activeHourStart.format(hourFormat), currentStartTime.format(hourFormat)));
+		}
+		
+		
+		if (isTimeBeforeActiveHour(activeHourEnd, currentEndTime)) {
+			return sb.toString();
+		}
+		
+		if (isTimeAfterActiveHour(activeHourStart, currentEndTime)) {
+			currentEndTime = activeHourStart;
+		}
+
+		
+		return getAllFreeSlot(currentEndTime, same_day, sb);
+
+	}
+	
+	private String getAllFreeSlot(LocalDateTime currentEndTime, int same_day, StringBuilder sb) {
```

Comment:
Do not use '_' in naming variables. It will complicate the readability of the code. Use only the camelCase.